base/datafmt.jl

```diff
@@ -119,6 +119,7 @@ readdlm_auto(input::Vector{UInt8}, dlm::Char, T::Type, eol::Char, auto::Bool; op
 readdlm_auto(input::IO, dlm::Char, T::Type, eol::Char, auto::Bool; opts...) =
     readdlm_string(read(input, String), dlm, T, eol, auto, val_opts(opts))
 function readdlm_auto(input::AbstractString, dlm::Char, T::Type, eol::Char, auto::Bool; opts...)
+    isfile(input) || error("Cannot read \'$input\': not a file")
```

Comment:
`ArgumentError` also seems reasonable :). I suggested `SystemError` on the basis of `open`'s behavior in the same case:
```julia
julia> open("foo")
ERROR: SystemError: opening file foo: No such file or directory
Stacktrace:
 [1] #systemerror#44 at ./error.jl:77 [inlined]
 [2] systemerror at ./error.jl:77 [inlined]
 [3] open(::String, ::Bool, ::Bool, ::Bool, ::Bool, ::Bool) at ./iostream.jl:104
 [4] open(::String) at ./iostream.jl:113
```